splash/request_middleware.py

```diff
@@ -72,6 +72,59 @@ def process(self, request, render_options, operation, data):
         return request
 
 
+class ContentTypeMiddleware(object):
+    """
+    Response middleware, aborts responses depending on the content type.
+    A response will be aborted (and the underlying connection closed) after
+    receiving the response headers if the content type of the response is not
+    in the whitelist or it's in the blacklist. Both lists support wildcards.
+    """
+    def __init__(self, verbosity=0):
+        self.verbosity = verbosity
+
+    @staticmethod
+    def contains(mime_set, mime):
+        for part1 in mime[0], '*':
+            for part2 in mime[1], '*':
+                if (part1, part2) in mime_set:
+                    return True
+        return False
+
+    @staticmethod
+    def split_mime(mime):
+        """
+        Split a mime string into type and subtype: 'text/html; charset=utf-8' -> ('text', 'html')
+        """
+        separator = mime.find(';')
+        if separator > 0:
+            mime = mime[:separator].strip()
+        parts = mime.split("/")
+        if len(parts) != 2:
+            return None
+        return parts[0], parts[1]
```

Comment:
Could you please make these static methods regular functions?
It'd be also nice to add doctests for them. For example, you can convert your existing example to a doctest which will run automatically:

```
Split a mime string into type and subtype

>>> split_mime('text/html; charset=utf-8')
('text', 'html')
```